<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Bazaar::Internal::PullOrPushDialog</class>
 <widget class="QDialog" name="Bazaar::Internal::PullOrPushDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>477</width>
    <height>388</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Branch Location</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QRadioButton" name="defaultButton">
        <property name="text">
         <string>Default location</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QRadioButton" name="localButton">
        <property name="text">
         <string>Local filesystem:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="Utils::PathChooser" name="localPathChooser">
        <property name="enabled">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QRadioButton" name="urlButton">
        <property name="toolTip">
         <string>For example: 'https://[user[:pass]@]host[:port]/[path]'.</string>
        </property>
        <property name="text">
         <string>Specify URL:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLineEdit" name="urlLineEdit">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>For example: 'https://[user[:pass]@]host[:port]/[path]'.</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Options</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0" colspan="2">
       <widget class="QCheckBox" name="rememberCheckBox">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Remember specified location as default</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0" colspan="2">
       <widget class="QCheckBox" name="overwriteCheckBox">
        <property name="toolTip">
         <string>Ignores differences between branches and overwrites
unconditionally.</string>
        </property>
        <property name="text">
         <string>Overwrite</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0" colspan="2">
       <widget class="QCheckBox" name="useExistingDirCheckBox">
        <property name="toolTip">
         <string>By default, push will fail if the target directory exists, but does not already have a control directory.
This flag will allow push to proceed.</string>
        </property>
        <property name="text">
         <string>Use existing directory</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0" colspan="2">
       <widget class="QCheckBox" name="createPrefixCheckBox">
        <property name="toolTip">
         <string>Creates the path leading up to the branch if it does not already exist.</string>
        </property>
        <property name="text">
         <string>Create prefix</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Revision:</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QLineEdit" name="revisionLineEdit"/>
      </item>
      <item row="2" column="0" colspan="2">
       <widget class="QCheckBox" name="localCheckBox">
        <property name="toolTip">
         <string>Performs a local pull in a bound branch.
Local pulls are not applied to the master branch.</string>
        </property>
        <property name="text">
         <string>Local</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>4</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Utils::PathChooser</class>
   <extends>QWidget</extends>
   <header location="global">utils/pathchooser.h</header>
   <container>1</container>
   <slots>
    <signal>editingFinished()</signal>
    <signal>browsingFinished()</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>Bazaar::Internal::PullOrPushDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>257</x>
     <y>177</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>Bazaar::Internal::PullOrPushDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>325</x>
     <y>177</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>urlButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>urlLineEdit</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>80</x>
     <y>121</y>
    </hint>
    <hint type="destinationlabel">
     <x>332</x>
     <y>123</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>localButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>localPathChooser</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>112</x>
     <y>81</y>
    </hint>
    <hint type="destinationlabel">
     <x>346</x>
     <y>81</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>urlButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>rememberCheckBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>71</x>
     <y>92</y>
    </hint>
    <hint type="destinationlabel">
     <x>163</x>
     <y>153</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>localButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>rememberCheckBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>71</x>
     <y>67</y>
    </hint>
    <hint type="destinationlabel">
     <x>163</x>
     <y>153</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
